pkg/webrtctransport.go

```diff
@@ -53,51 +57,79 @@ func NewWebRTCTransport(ctx context.Context, session *Session, me MediaEngine, c
 		pc:      pc,
 		me:      me,
 		session: session,
-		routers: make(map[uint32]*Router),
+		routers: make(map[string]Router),
+		label:   cuid.New(),
 	}
 
 	// Subscribe to existing transports
 	for _, t := range session.Transports() {
 		for _, router := range t.Routers() {
-			sender, err := p.NewSender(router.Track())
-			log.Infof("Init add router ssrc %d to %s", router.Track().SSRC(), p.id)
+			err := router.AddSender(p)
+			// log.Infof("Init add router ssrc %d to %s", router.receivers[0].Track().SSRC(), p.id)
 			if err != nil {
-				log.Errorf("Error subscribing to router %v", router)
+				log.Errorf("Error subscribing to router err: %v", err)
 				continue
 			}
-			router.AddSender(p.id, sender)
 		}
 	}
 
 	// Add transport to the session
 	session.AddTransport(p)
 
 	pc.OnTrack(func(track *webrtc.Track, receiver *webrtc.RTPReceiver) {
-		log.Debugf("Peer %s got remote track id: %s ssrc: %d", p.id, track.ID(), track.SSRC())
-		recv := NewWebRTCReceiver(ctx, track)
+		log.Debugf("Peer %s got remote track id: %s ssrc: %d rid :%s label: %s", p.id, track.ID(), track.SSRC(), track.RID(), track.Label())
+		recv := NewWebRTCReceiver(ctx, track, cfg.router)
 
 		if recv.Track().Kind() == webrtc.RTPCodecTypeVideo {
 			go p.sendRTCP(recv)
 		}
-
-		router := NewRouter(p.id, recv)
-		log.Debugf("Created router %s %d", p.id, recv.Track().SSRC())
-
-		p.session.AddRouter(router)
+		if router, ok := p.routers[track.ID()]; !ok {
+			if track.RID() != "" {
+				router = newRouter(p.id, cfg.router, SimulcastRouter)
+				go func() {
+					// Send 3 big remb msgs to fwd all the tracks
```

Comment:
So we dont do any congestion control on the pub tracks right now?

Reply:
yes, currently a remb loop is called per track, to calculate the bandwidth and later downgrade senders, but we can omit fwding the remb packet.